src/company_cmd.cpp

```diff
@@ -1179,3 +1180,54 @@ uint32 CompanyInfrastructure::GetTramTotal() const
 	}
 	return total;
 }
+
+void CcGiveMoney(const CommandCost &result, TileIndex tile, uint32 p1, uint32 p2, uint32 cmd)
+{
+	if (result.Failed() || !_settings_game.economy.give_money) return;
+
+	/* Inform the company of the action of one of its clients (controllers). */
+	char msg[64];
+	SetDParam(0, p2);
+	GetString(msg, STR_COMPANY_NAME, lastof(msg));
+
+	if (!_network_server) {
+		NetworkClientSendChat(NETWORK_ACTION_GIVE_MONEY, DESTTYPE_TEAM, p2, msg, p1);
+	} else {
+		NetworkServerSendChat(NETWORK_ACTION_GIVE_MONEY, DESTTYPE_TEAM, p2, msg, CLIENT_ID_SERVER, p1);
+	}
```

Comment:
Possibly not in scope of this PR:
* Why does the client have to send a chat message to tell others about transfering money?
* The client could block that.
* Every client already receives the command, so could display a message locally in the command.